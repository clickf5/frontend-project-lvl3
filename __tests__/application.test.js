import { promises as fs } from 'fs';
import path from 'path';
import { html } from 'js-beautify';
import userEvent from '@testing-library/user-event';
import timer from 'timer-promise';

import app from '../src/js/application';

const htmlOptions = {
  preserve_newlines: false,
  unformatted: [],
};

const fixturesPath = path.join(__dirname, '__fixtures__');
const getTree = () => html(document.body.innerHTML, htmlOptions);

let elements;

beforeEach((done) => {
  fs.readFile(path.join(fixturesPath, 'index.html')).then((data) => {
    const initHTML = data.toString();
    document.documentElement.innerHTML = initHTML;
    app();
    elements = {
      url: document.querySelector('[name="url"]'),
    };
    done();
  });
});

test('init and wrong url', () => {
  expect(getTree()).toMatchSnapshot();
  return Promise
    .resolve()
    .then(() => userEvent.type(elements.url, 'wrong url', { allAtOnce: true }))
    .then(() => elements.url.setAttribute('value', 'wrong url'))
    .then(() => timer.start(10))
    .then(() => expect(getTree()).toMatchSnapshot());
});

test('valid url', () => Promise
  .resolve()
  .then(() => userEvent.type(elements.url, 'https://vc.ru/rss', { allAtOnce: true }))
  .then(() => elements.url.setAttribute('value', 'https://vc.ru/rss'))
  .then(() => timer.start(10))
  .then(() => expect(getTree()).toMatchSnapshot()));

test('required url', () => Promise
  .resolve()
  .then(() => userEvent.type(elements.url, '', { allAtOnce: true }))
  .then(() => elements.url.setAttribute('value', ''))
  .then(() => timer.start(10))
  .then(() => expect(getTree()).toMatchSnapshot()));
