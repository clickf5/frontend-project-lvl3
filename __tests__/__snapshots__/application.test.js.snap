// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init and wrong url 1`] = `
"<div class=\\"container mt-5\\">
    <div class=\\"jumbotron\\">
        <h1>RSS Reader</h1>
        <p class=\\"lead\\">Start reading RSS today! It is easy, it is nicely.</p>
        <form action=\\"\\" class=\\"form-inline rss-form\\">
            <div class=\\"form-group\\">
                <input type=\\"text\\" required=\\"\\" name=\\"url\\" class=\\"form-control\\" placeholder=\\"rss link\\">
            </div>
            <button type=\\"submit\\" class=\\"btn btn-primary ml-1\\">Add</button>
        </form>
        <div class=\\"feedback\\"></div>
    </div>
    <div class=\\"row\\">
        <div class=\\"col rss-items\\"></div>
        <div class=\\"col rss-links\\"></div>
    </div>
    <footer class=\\"footer\\">
        <p>© Eugene Panin</p>
    </footer>
</div> <!-- container -->
<script src=\\"./js/main.js\\"></script>"
`;

exports[`init and wrong url 2`] = `
"<div class=\\"container mt-5\\">
    <div class=\\"jumbotron\\">
        <h1>RSS Reader</h1>
        <p class=\\"lead\\">Start reading RSS today! It is easy, it is nicely.</p>
        <form action=\\"\\" class=\\"form-inline rss-form\\">
            <div class=\\"form-group\\">
                <input type=\\"text\\" required=\\"\\" name=\\"url\\" class=\\"form-control is-invalid\\" placeholder=\\"rss link\\" value=\\"wrong url\\">
            </div>
            <button type=\\"submit\\" class=\\"btn btn-primary ml-1\\" disabled=\\"\\">Add</button>
        </form>
        <div class=\\"feedback text-danger\\">Field value must be a valid URL</div>
    </div>
    <div class=\\"row\\">
        <div class=\\"col rss-items\\"></div>
        <div class=\\"col rss-links\\"></div>
    </div>
    <footer class=\\"footer\\">
        <p>© Eugene Panin</p>
    </footer>
</div> <!-- container -->
<script src=\\"./js/main.js\\"></script>"
`;

exports[`required url 1`] = `
"<div class=\\"container mt-5\\">
    <div class=\\"jumbotron\\">
        <h1>RSS Reader</h1>
        <p class=\\"lead\\">Start reading RSS today! It is easy, it is nicely.</p>
        <form action=\\"\\" class=\\"form-inline rss-form\\">
            <div class=\\"form-group\\">
                <input type=\\"text\\" required=\\"\\" name=\\"url\\" class=\\"form-control is-invalid\\" placeholder=\\"rss link\\" value=\\"\\">
            </div>
            <button type=\\"submit\\" class=\\"btn btn-primary ml-1\\" disabled=\\"\\">Add</button>
        </form>
        <div class=\\"feedback text-danger\\">Field is required</div>
    </div>
    <div class=\\"row\\">
        <div class=\\"col rss-items\\"></div>
        <div class=\\"col rss-links\\"></div>
    </div>
    <footer class=\\"footer\\">
        <p>© Eugene Panin</p>
    </footer>
</div> <!-- container -->
<script src=\\"./js/main.js\\"></script>"
`;

exports[`valid url 1`] = `
"<div class=\\"container mt-5\\">
    <div class=\\"jumbotron\\">
        <h1>RSS Reader</h1>
        <p class=\\"lead\\">Start reading RSS today! It is easy, it is nicely.</p>
        <form action=\\"\\" class=\\"form-inline rss-form\\">
            <div class=\\"form-group\\">
                <input type=\\"text\\" required=\\"\\" name=\\"url\\" class=\\"form-control\\" placeholder=\\"rss link\\" value=\\"https://vc.ru/rss\\">
            </div>
            <button type=\\"submit\\" class=\\"btn btn-primary ml-1\\">Add</button>
        </form>
        <div class=\\"feedback\\"></div>
    </div>
    <div class=\\"row\\">
        <div class=\\"col rss-items\\"></div>
        <div class=\\"col rss-links\\"></div>
    </div>
    <footer class=\\"footer\\">
        <p>© Eugene Panin</p>
    </footer>
</div> <!-- container -->
<script src=\\"./js/main.js\\"></script>"
`;
